app/microsoft/user.py

```diff
@@ -0,0 +1,56 @@
+from microsoft.base import BaseEndpoint
+from microsoft.helpers import generate_password
+
+
+class UserEndpoint(BaseEndpoint):
+    """
+    Endpoint for User.
+    https://docs.microsoft.com/en-us/graph/api/resources/user
+    """
+
```

Comment:
probably better to make this argument "email" than "userPrincipalName"
no action required

Reply:
also prefer snake_case over camelCase

Reply:
right ... I see what you mean.

the use of userPrincipalName is a bit odd stylistically, the reason I have used it is for consistency with the documentation (https://docs.microsoft.com/en-us/graph/api/user-get?view=graph-rest-1.0&tabs=http), would like the API request to be easy to map to the documentation example

I am worried that using param like 'email' might create some confusion, in the MSGraph concepts, 'email' and 'userPrincipalName' are different things, though they might be the same value.